### PR TITLE
Feat/config file

### PR DESCRIPTION
**Pull request by mathild3r**
Add config file:

Config should be at root
If argument are present as well as config file option, argument override config file

Todo later: change label format and add new option

If config file, parse the yaml file and use the ifThenElsexxx function to set the pointer
If then else function: 
  main purpose is to do a comparison of int in one line
  check if the flag is present if not check if value in config file exist
  return default otherwise

For tests purposes the flags need to be resetted otherwise they keep the value of the precedent run.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Use the binaries from [the release page](https://github.com/snyk-tech-services/j
     -dryRun=<true|false>                                                // Optional. result can be found in a json file were the tool is run
     -debug=<true|false>                                                 // Optional. enable debug mode
     -ifUpgradeAvailableOnly=<true|false>                                // Optional. create ticket only for upgradable issues
+    -configFile=<true|false>                                            // Optional. Use jira.yaml to set the config
```
Comment (lili2311) [inline README.md:41]: should we take the path to file here instead? Seems odd to have this as a boolean and then read a file from dick. Usually you would provide the path to file `--configFile=jira.yaml`

Comment (aarlaud) [reply README.md:41]: agreed

File: fixtures/jiraConfig.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+schema: 1
+snyk: 
+    orgID: a1b2c3de-99b1-4f3f-bfdb-6ee4b4990513
```
[the remark this answers is not in the document]
Comment (aarlaud) [reply fixtures/jiraConfig.yaml:3]: Need README entry indeed for the file format specifically yes. But what field means what is already in the README iirc.

File: jira_test.go [hunk 3]
```diff
@@ -700,3 +701,133 @@ func TestOpenJiraTicketDryRyn(t *testing.T) {
...
+	options := flags{}
+	options.setOption()
+
+	// Moving the config file back
+	err = os.Rename("./jira.yaml", "./fixtures/jiraConfigV2.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	assert.Equal(options.mandatoryFlags.orgID, "456")
```
Comment (aarlaud) [inline jira_test.go:767]: I'd suggest to have a logic to test all flags in each mandatory and optional flags structs. Makes it easier to then just focus on updating the fixture(s)

File: utils.go [hunk 4]
```diff
@@ -89,43 +92,135 @@ type optionalFlags struct {
 	ifUpgradeAvailableOnly bool
 }
 
+type snyk struct {
```
Comment (aarlaud) [inline utils.go:95]: should define those types outside of utils.go, they are too burried here

File: utils.go [hunk 5]
```diff
@@ -57,75 +53,99 @@ func (m *debug) Printf(format string, args ...interface{}) {
...
-	Of.priorityIsSeverity = *priorityIsSeverityPtr
-	Of.priorityScoreThreshold = *priorityScorePtr
-	Of.debug = *debugPtr
-	Of.dryRun = *dryRunPtr
-	Of.ifUpgradeAvailableOnly = *ifUpgradeAvailableOnlyPtr
+	maturityFilterPtr *string, typePtr *string, assigneeNamePtr *string,
+	assigneeIDPtr *string, labelsPtr *string, priorityIsSeverityPtr *bool,
+	priorityScorePtr *int, debugPtr *bool, dryRunPtr *bool, ifUpgradeAvailableOnlyPtr *bool, config *config) {
+
+	if config != nil {
```
Comment (aarlaud) [inline utils.go:88]: this needs refactoring

**Review by aarlaud — APPROVED** [review]

**Pull request merged**